design/rest-api-documentation.md

```diff
@@ -0,0 +1,133 @@
+
+```
+public class AccountRestService {
+
+    ...
+
+  @Path("/applications")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Operation(description = "List of applications")
+  @APIResponses(value = {
+  	@APIResponse(
+  			responseCode = "200",
+  			description = "Successful, List of applications"
+      ),
+  	@APIResponse(
+  			responseCode = "401",
+  			description = "Unauthorized, Failure"
+      ),
+  	@APIResponse(
+  			responseCode = "500",
+  			description = "Error in getting the list of applications"
+      )
+  })
+  public Response applications(@QueryParam("name") String name) {
+  	...
+  }
+
+}
+```
+
+A new dependency must be introduced in the Keycloak `pom.xml` file:
+
+```
+<dependency>
+    <groupId>org.eclipse.microprofile.openapi</groupId>
+    <artifactId>microprofile-openapi-api</artifactId>
+</dependency>
+```
+
+By spinning up the Keycloak server, developers should be able to send an HTTP request to http://localhost:8080/auth/openapi and get the OpenAPI document. Here is an example:
+
+```
+$ curl http://localhost:8080/auth/openapi
+---
+openapi: 3.0.1
+info:
+  title: Account REST API
+  contact:
+    name: Keycloak team
+    url: https://keycloak.org
+    email: keycloak-dev@googlegroups.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: v1alpha1
+servers:
+- url: /auth
+tags:
+- name: Account REST API
+  description: Account REST API operations.
+- name: OpenAPI Example
+  description: Get a text in various formats
+- name: Account REST service
+  description: REST API for the Account REST service
+paths:
+  /account/applications:
+    get:
+      tags:
+      - Account REST API
+      description: List of applications
+      parameters:
+      - name: name
+        in: query
+        schema:
+          type: string
+      responses:
+        "401":
+          description: Unauthorized, Failure
+        "500":
+          description: Error in getting the list of applications
+        "200":
+          description: Successful, List of applications
+```
+
+As we can see, with minimal effort it is possible to produce a JSON document that describes the functionalities of our API and can be used as a foundation for UI tools like Swagger to visualize and interact with our APIs. This document is served as a static file.
+
+## Swagger UI
+
+Swagger UI allows developers to visualize and generate documentation based on the OpenAPI specification. There are several ways of doing this:
+
+1. Inserting the URL in the [Swagger generator](https://generator.swagger.io/) which is available online. For example: http://localhost:8080/auth/openapi
+2. Spinning up a Docker image: 
+```
+docker pull swaggerapi/swagger-ui
+docker run -p 80:8080 swaggerapi/swagger-ui
+```   
+3. [Swagger Codegen](https://github.com/swagger-api/swagger-codegen#building). The best way to generate static HTML files for documentation.
```

Comment:
I would actually recommend this instead:  https://github.com/OpenAPITools/openapi-generator

Although again, I think ReDoc is a better experience for users consuming the documentation, but that's a matter of preference (ReDoc has its own CLI for generating static/publishable HTML).

Reply:
Thanks Eric, I will look into that and update the document.